src/helpers/month-data-array.model.ts

```diff
@@ -0,0 +1,28 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { Opaque } from "../utils/type-utils";
+
+export interface MonthDataArray<T> extends Opaque<"MonthData", T[]> {
+  map<U>(
+    callbackfn: (value: T, index: number, array: T[]) => U,
+    thisArg?: unknown
+  ): MonthDataArray<U>;
+  filter<S extends T>(
+    predicate: (value: T, index: number, array: T[]) => value is S,
+    thisArg?: unknown
+  ): MonthDataArray<S>;
+  filter(
+    predicate: (value: T, index: number, array: T[]) => unknown,
+    thisArg?: unknown
+  ): MonthDataArray<T>;
+  every<S extends T>(
+    predicate: (value: T, index: number, array: T[]) => value is S,
+    thisArg?: unknown
+  ): this is MonthDataArray<S>;
+  every(predicate: (value: T, index: number, array: T[]) => unknown, thisArg?: unknown): boolean;
+  splice(start: number, deleteCount?: number): MonthDataArray<T>;
+  splice(start: number, deleteCount: number, ...items: T[]): MonthDataArray<T>;
+  slice(start?: number, end?: number): MonthDataArray<T>;
+  reverse(): MonthDataArray<T>;
```

Comment:
Wprowadziłem `Opaque` na typ odpowiadający danym z miesiąca.
Niestety musiałem przekopiować tutaj niektóre deklaracje metod z `Array` ponieważ one wcześnie zwracały `T[]` zamiast `MonthDataArray<T>` i potrzeba było to podmienić.
Jeśli masz lepszy pomysł jak to można ugryźć to chętnie wysłucham i zaaplikuję